src/basic/ordered-set.h

```diff
@@ -75,6 +75,18 @@ void ordered_set_print(FILE *f, const char *field, OrderedSet *s);
 #define ORDERED_SET_FOREACH(e, s) \
         _ORDERED_SET_FOREACH(e, s, UNIQ_T(i, UNIQ))
 
+#define ordered_set_clear_with_destructor(_s, _f)               \
+        ({                                                      \
+                void *_item;                                    \
+                while ((_item = ordered_set_steal_first(_s)))   \
```

Comment:
hmm, this evals `_s` once in every loop. Maybe add a local variable you assign the object first to, and then use that.